pkg/controllers/provisioning/scheduling/topology.go

```diff
@@ -394,13 +394,29 @@ func TopologyListOptions(namespace string, labelSelector *metav1.LabelSelector)
 		selector = selector.Add(*requirement)
 	}
 	for _, expression := range labelSelector.MatchExpressions {
-		requirement, err := labels.NewRequirement(expression.Key, selection.Operator(expression.Operator), expression.Values)
+		requirement, err := labels.NewRequirement(expression.Key, mapOperator(expression.Operator), expression.Values)
```

Comment:
This branch added the runtime.Must() to the result of NewRequirement, previously we just ignored the error assuming it couldn't happen.  It did, and we ended up just ignoring all match expressions.